rfc/rfc013-vc-irma-proof.md

```diff
@@ -0,0 +1,272 @@
+The `message` is used to connect the signatures and its attributes to a DID. This is the actual text the user will sign.
+The `A` fields identifies which credential is used from the IRMA scheme.
+The `a_disclosed` field contains the actual attribute values. These values are base64 representations of big integers. Big integers are required for the cryptography to work.
+Underneath they contain strings, booleans, integers or enums.
+All the other fields are required to validate the authenticity of the signature. 
+
+The example signature from above will be base64 encoded when used in a Verifiable Credential proof:
+
+Below is a complete example of how an Irma proof can be embedded in a Verifiable Credential:
+
+```json
+{
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://nuts.nl/credentials/v1"
+  ],
+  "id": "did:nuts:t1DVVAs5fmNba8fdKoTSQNtiGcH49vicrkjZW2KRqpv#6f91673b-afa9-4d26-9e0f-00d989943275",
+  "issuanceDate": "2021-03-15T16:34:17.687862+01:00",
+  "issuer": "did:nuts:t1DVVAs5fmNba8fdKoTSQNtiGcH49vicrkjZW2KRqpv",
+  "type": ["VerifiableCredential", "NutsKVKCredential"],
+  "credentialSubject": {
+    "id": "did:nuts:t1DVVAs5fmNba8fdKoTSQNtiGcH49vicrkjZW2KRqpv",
+    "pbdf": {
+      "kvk": {
+        "official": {
+          "legalEntity": "CareBears",
+          "officeAddress": "CarePlace",
+          "kvkNumber": "12345678"
+        }
+      }
+    }
+  },
+  "proof": {
+    "type": "NutsIrmaCredentialProof",
+    "proofValue": "DgYdYMUYHURJLD7xdnWRinqWCEY5u5fK...j915Lt3hMzLHoPiPQ9sSVfRrs1D"
+  }
+}
+```
+
+### 4.2 Verification
+
+Verification of an IRMA signature MUST be done by an [IRMA server](https://irma.app/docs/irma-server/) instance. 
+It's also possible for an implementing party to embed the [IRMA go libraries](https://github.com/privacybydesign/irmago).
+The verification result will indicate if the signature is valid.
+The result will also contain the disclosed attributes as dot-delimited strings.
+All attributes in the `credentialSubject` MUST be present in the result attributes.
+The dot-delimited format of the verification result can be extended as follows:
+
+```text
+A.B.C.D=X
+```
+
+becomes
+
+```json
+{
+  "A": {
+    "B": {
+      "C": {
+        "D": "X"
+      }
+    }
+  }
+}
+```
+
+The last validation is to check if the `message` from the IRMA signature matches the value of `credentialSubject.id`.
+
+### 4.3 IRMA scheme
+
+The IRMA ecosystem has two different schemes: `pbdf` and `irma-demo`. The latter one is not intended for production since all private keys are published with the scheme. Any implementing party MUST ignore all signatures made with that scheme in production.
+
+The contents of the schemes can be viewed on Github: [pbdf](https://github.com/privacybydesign/pbdf-schememanager) and [irma-demo](https://github.com/privacybydesign/irma-demo-schememanager)
+
+### 4.4 Revocation
+
+Although IRMA supports revocation for credentials, this isn't usable for digital signatures.
```

Comment:
I'm missing how IRMA revocation relates to VC revocation (and moreover, this RFC in general). If unrelated, it's still checked when verifying the IRMA signature (by the IRMA implementation). If related, specify how it relates to VC revocation.

An IRMA VC Proof can be revoked while the IRMA credential isn't (e.g. theft of subject DID private key). An IRMA VC Proof can be active (unrevoked) while the IRMA credential is revoked (IRMA issuer private key stolen).

Reply:
That's what the beginning of §4 is trying to tell you. IRMA revocation can only be checked during issuance. I'll add a passage about personal responsibility when an IRMA credential is revoked.